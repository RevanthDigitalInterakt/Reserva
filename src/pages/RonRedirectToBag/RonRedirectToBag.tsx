import React, { useCallback, useEffect, useState } from 'react';
import type { StackScreenProps } from '@react-navigation/stack';

import { URLSearchParams } from 'react-native-url-polyfill';
import { SafeAreaView } from 'react-native';
import { useCart } from '../../context/CartContext';
import type { RootStackParamList } from '../../routes/StackNavigator';
import EventProvider from '../../utils/EventProvider';
import useAsyncStorageProvider from '../../hooks/useAsyncStorageProvider';
import { adaptOrderFormItemsTrack } from '../../utils/adaptOrderFormItemsTrack';
import { RonRedirectTypeEnum, useRonRedirectLazyQuery } from '../../base/graphql/generated';
import { urlHandler } from '../../config/linking';
import { TopBarBackButton } from '../../modules/Menu/components/TopBarBackButton';
import { LoadingScreen } from '../../common/components/LoadingScreen';

export type IRonRedirectToBagProps = StackScreenProps<RootStackParamList, 'RonRedirectToBag'>;

export default function RonRedirectToBag({ route, navigation }: IRonRedirectToBagProps) {
  const { ronCode } = route?.params || {};
  const { topBarLoading, orderForm, restoreCart } = useCart();
  const { setItem } = useAsyncStorageProvider();
  const [getRonRedirect] = useRonRedirectLazyQuery({ context: { clientName: 'gateway' } });
  const [finished, setFinished] = useState(false);

  const saveOrderFormItems = useCallback(async (orderFormId: string) => {
    const productIds = new Set((orderForm?.items || []).map((item) => item.productId));

    await setItem('@RNOrder:RonItems', Array.from(productIds));
    await setItem('@RNSession:Ron', true);

    EventProvider.logEvent(
      'ron_open',
      {
        open: ronCode,
        items: adaptOrderFormItemsTrack(orderForm?.items),
      },
    );

    navigation.replace(
      'BagScreen',
      { isProfileComplete: false, orderFormId },
    );
  }, [setItem, orderForm, ronCode]);

  const handleCustomRedirect = useCallback(async (url: string) => {
    const deepLinkUrl = await urlHandler(url);
    const searchParams = new URLSearchParams(deepLinkUrl.split('?')[1]);
    const dynamicParams: { [key: string]: string } = {};

    searchParams.forEach((value, key) => {
      dynamicParams[key] = value;
    });

    if (!dynamicParams) return;

    const shouldNavigateToProductDetail = dynamicParams?.skuId;

    if (shouldNavigateToProductDetail) {
      navigation.replace('AsyncDeepLink', {
        reducerKey: 'PRODUCT',
        ...dynamicParams,
      });
    }

    const shouldNavigateToCatalog = dynamicParams?.params;

    if (shouldNavigateToCatalog) {
      navigation.replace('AsyncDeepLink', {
        reducerKey: 'CATALOG',
        ...dynamicParams,
      });
    }
  }, [navigation]);

  const handleRedirect = useCallback(async (code: string) => {
    if (!code) return;

    const { data, error } = await getRonRedirect({ variables: { code: ronCode } });

    if (!data?.ronRedirect || error) {
      navigation.replace('Home');
      return;
    }

    const { orderFormId, url, type: redirectType } = data.ronRedirect;

    if (redirectType === RonRedirectTypeEnum.Orderform && orderFormId) {
      await restoreCart(orderFormId);
      await saveOrderFormItems(orderFormId);
      return;
    }

    if (redirectType === RonRedirectTypeEnum.Custom && url) {
      handleCustomRedirect(url);
    }
  }, [saveOrderFormItems, handleCustomRedirect, getRonRedirect, ronCode, restoreCart, navigation]);

  useEffect(() => {
    if (!finished && ronCode) {
      setFinished(true);
      handleRedirect(ronCode);
    }
  }, [ronCode, handleRedirect, finished]);

  return (
    <SafeAreaView style={{ justifyContent: 'space-between', flex: 1, backgroundColor: '#fff' }}>
      <TopBarBackButton showShadow loading={topBarLoading} />

      <LoadingScreen />
    </SafeAreaView>
  );
}
