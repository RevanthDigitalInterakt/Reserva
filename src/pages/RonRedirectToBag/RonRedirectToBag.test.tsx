import React from 'react';
import { MockedProvider } from '@apollo/client/testing';
import { ThemeProvider } from 'styled-components/native';
import type { DocumentNode } from 'graphql';
import AsyncStorageMock from '@react-native-async-storage/async-storage/jest/async-storage-mock';
import
{
  cleanup,
  render,
  waitFor,
} from '@testing-library/react-native';
import AsyncStorage from '@react-native-async-storage/async-storage';
import RonRedirectToBag from './RonRedirectToBag';
import { CartContext } from '../../context/CartContext';
import { RonRedirectDocument, type RonRedirectQuery, RonRedirectTypeEnum } from '../../base/graphql/generated';
import { theme } from '../../base/usereservappLegacy/theme';

interface IApolloMock<T> {
  request: {
    query: DocumentNode;
    variables: object;
  };
  result: { data: T; }
}

const RON_ORDERFORM_CODE = '0NxiK7FA80gk';
const RON_REDIRECT_PRODUCT_CODE = '1MiUK4FB16op';
const RON_REDIRECT_CATALOG_CODE = '123';

const RON_INVALID_CODE = '1234567890';

const navigateFn = jest.fn();
const replaceFn = jest.fn();
const navigationMock = { navigate: navigateFn, replace: replaceFn };

const apolloMocks: Array<IApolloMock<RonRedirectQuery>> = [
  {
    request: {
      query: RonRedirectDocument,
      variables: { code: RON_INVALID_CODE },
    },
    result: {
      data: {
        ronRedirect: null,
        __typename: 'Query',
      },
    },
  },
  {
    request: {
      query: RonRedirectDocument,
      variables: { code: RON_ORDERFORM_CODE },
    },
    result: {
      data: {
        ronRedirect: {
          orderFormId: 'f30f6121fa6543e8a288965e6dbcd7fe',
          url: '',
          type: RonRedirectTypeEnum.Orderform,
        },
        __typename: 'Query',
      },
    },
  },
  {
    request: {
      query: RonRedirectDocument,
      variables: { code: RON_REDIRECT_PRODUCT_CODE },
    },
    result: {
      data: {
        ronRedirect: {
          orderFormId: '',
          url: 'https://www.usereserva.com/peitoral-mesh-flag0079448/p?skuId=425418',
          type: RonRedirectTypeEnum.Custom,
        },
        __typename: 'Query',
      },
    },
  },
  {
    request: {
      query: RonRedirectDocument,
      variables: { code: RON_REDIRECT_CATALOG_CODE },
    },
    result: {
      data: {
        ronRedirect: {
          orderFormId: '',
          url: 'https://www.usereserva.com/peitoral-mesh-flag0079448/p?params=123',
          type: RonRedirectTypeEnum.Custom,
        },
        __typename: 'Query',
      },
    },
  },
];

jest.mock('../../zustand/useApolloFetchPolicyStore', () => ({
  useApolloFetchPolicyStore: () => ({
    getFetchPolicyPerKey: jest.fn(),
  }),
}));

AsyncStorageMock.setItem = jest.fn();

describe('RonRedirectToBag', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  afterEach(() => {
    jest.clearAllTimers();
  });

  it('renders without error and match snapshot', () => {
    function MockedCartContext({ children }) {
      return (
        <CartContext.Provider value={{
          orderform: jest.fn().mockResolvedValue({ orderFormId: '128adb08596442708ee89e2a0f561321' }),
        }}
        >
          {children}
        </CartContext.Provider>
      );
    }

    const ComponentForSnapShot = (
      <ThemeProvider theme={theme}>
        <MockedProvider addTypename={false} mocks={apolloMocks}>
          <MockedCartContext>
            <RonRedirectToBag
              navigation={navigationMock as any}
              route={{
                params: { ronCode: '' },
                name: 'RonRedirectToBag',
                key: '',
              }}
            />
          </MockedCartContext>
        </MockedProvider>
      </ThemeProvider>
    );

    const root = render(ComponentForSnapShot);
    expect(root.toJSON()).toMatchSnapshot();
    cleanup();
  });

  it('must load a ron with orderform code and setItem', async () => {
    function MockedCartContext({ children }) {
      return (
        <CartContext.Provider value={{
          orderform: jest.fn().mockResolvedValue({ orderFormId: '128adb08596442708ee89e2a0f561321' }),
          restoreCart: jest.fn().mockResolvedValue({}),
        }}
        >
          {children}
        </CartContext.Provider>
      );
    }

    const ComponentForSnapShot = (
      <ThemeProvider theme={theme}>
        <MockedProvider addTypename={false} mocks={apolloMocks}>
          <MockedCartContext>
            <RonRedirectToBag
              navigation={navigationMock as any}
              route={{
                params: { ronCode: RON_ORDERFORM_CODE },
                name: 'RonRedirectToBag',
                key: '',
              }}
            />
          </MockedCartContext>
        </MockedProvider>
      </ThemeProvider>
    );

    render(ComponentForSnapShot);

    await waitFor(() => expect(AsyncStorage.setItem).toHaveBeenCalled());
    expect(replaceFn).toHaveBeenCalledTimes(1);
    expect(replaceFn).toHaveBeenCalledWith(
      'BagScreen',
      { isProfileComplete: false, orderFormId: 'f30f6121fa6543e8a288965e6dbcd7fe' },
    );
    cleanup();
  });

  it('must load a ron with redirect home', async () => {
    function MockedCartContext({ children }) {
      return (
        <CartContext.Provider value={{
          orderform: jest.fn().mockResolvedValue({ orderFormId: '' }),
          restoreCart: jest.fn().mockResolvedValue({}),
        }}
        >
          {children}
        </CartContext.Provider>
      );
    }

    const ComponentForSnapShot = (
      <ThemeProvider theme={theme}>
        <MockedProvider addTypename={false} mocks={apolloMocks}>
          <MockedCartContext>
            <RonRedirectToBag
              navigation={navigationMock as any}
              route={{
                params: { ronCode: RON_INVALID_CODE },
                name: 'RonRedirectToBag',
                key: '',
              }}
            />
          </MockedCartContext>
        </MockedProvider>
      </ThemeProvider>
    );

    await waitFor(() => render(ComponentForSnapShot));
    expect(replaceFn).toHaveBeenCalledWith('Home');
  });

  it('must load a ron with product url', async () => {
    function MockedCartContext({ children }) {
      return (
        <CartContext.Provider value={{
          orderform: jest.fn().mockResolvedValue({ orderFormId: '128adb08596442708ee89e2a0f561321' }),
          restoreCart: jest.fn().mockResolvedValue({}),
        }}
        >
          {children}
        </CartContext.Provider>
      );
    }

    const ComponentForSnapShot = (
      <ThemeProvider theme={theme}>
        <MockedProvider addTypename={false} mocks={apolloMocks}>
          <MockedCartContext>
            <RonRedirectToBag
              navigation={navigationMock as any}
              route={{
                params: { ronCode: RON_REDIRECT_PRODUCT_CODE },
                name: 'RonRedirectToBag',
                key: '',
              }}
            />
          </MockedCartContext>
        </MockedProvider>
      </ThemeProvider>
    );

    render(ComponentForSnapShot);

    await waitFor(() => expect(replaceFn).toHaveBeenCalledWith('AsyncDeepLink', {
      reducerKey: 'PRODUCT',
      skuId: '425418',
    }));
  });

  it('must load a ron with catalog url', async () => {
    function MockedCartContext({ children }) {
      return (
        <CartContext.Provider value={{
          orderform: jest.fn().mockResolvedValue({ orderFormId: '128adb08596442708ee89e2a0f561321' }),
          restoreCart: jest.fn().mockResolvedValue({}),
        }}
        >
          {children}
        </CartContext.Provider>
      );
    }

    const ComponentForSnapShot = (
      <ThemeProvider theme={theme}>
        <MockedProvider addTypename={false} mocks={apolloMocks}>
          <MockedCartContext>
            <RonRedirectToBag
              navigation={navigationMock as any}
              route={{
                params: { ronCode: RON_REDIRECT_CATALOG_CODE },
                name: 'RonRedirectToBag',
                key: '',
              }}
            />
          </MockedCartContext>
        </MockedProvider>
      </ThemeProvider>
    );

    render(ComponentForSnapShot);

    await waitFor(() => expect(replaceFn).toHaveBeenCalledWith('AsyncDeepLink', {
      reducerKey: 'CATALOG',
      params: '123',
    }));
  });
});
