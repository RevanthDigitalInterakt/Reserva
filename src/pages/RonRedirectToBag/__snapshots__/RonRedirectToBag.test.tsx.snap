// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RonRedirectToBag renders without error and match snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "space-between",
    }
  }
>
  <View
    bg="white"
    boxShadow="topBarShadow"
    height={50}
    justifyContent="flex-end"
    paddingX="quarck"
    style={
      Array [
        Object {
          "backgroundColor": "#FFF",
          "height": 50,
          "justifyContent": "flex-end",
          "paddingLeft": 4,
          "paddingRight": 4,
          "shadowColor": "rgba(0,0,0,0.1)",
          "shadowOffset": Object {
            "height": 3,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 2,
        },
        Object {
          "elevation": 10,
        },
      ]
    }
  >
    <View
      flex={1}
      flexDirection="row"
      justifyContent="flex-end"
      paddingX="micro"
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexDirection": "row",
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "flex-end",
            "paddingLeft": 12,
            "paddingRight": 12,
          },
        ]
      }
    >
      <View
        alignItems="flex-start"
        alignSelf="center"
        flexDirection="row"
        style={
          Array [
            Object {
              "alignItems": "flex-start",
              "alignSelf": "center",
              "flexDirection": "row",
              "width": "25%",
            },
          ]
        }
        width="25%"
      >
        <View
          alignItems="center"
          style={
            Array [
              Object {
                "alignItems": "center",
              },
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            hitSlop={
              Object {
                "bottom": 20,
                "left": 20,
                "right": 20,
                "top": 20,
              }
            }
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "opacity": 1,
              }
            }
            testID="com.usereserva:id/top_bar_button_go_back"
          >
            <View
              alignItems="center"
              flexDirection="row"
              justifyContent="center"
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                ]
              }
            >
              <RNSVGSvgView
                bbHeight={24}
                bbWidth={24}
                color="preto"
                focusable={false}
                onPress={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                responsible={true}
                size={24}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    Array [
                      Object {
                        "color": "#333333",
                        "height": 24,
                        "width": 24,
                      },
                    ],
                    Object {
                      "flex": 0,
                      "height": 24,
                      "width": 24,
                    },
                  ]
                }
                testID="com.usereserva:id/top_bar_button_go_back"
              >
                <RNSVGGroup>
                  <RNSVGSvgView
                    align="xMidYMid"
                    bbHeight="100%"
                    bbWidth="100%"
                    focusable={false}
                    height="100%"
                    meetOrSlice={0}
                    minX={0}
                    minY={0}
                    style={
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                          "borderWidth": 0,
                        },
                        Array [
                          Object {
                            "height": "100%",
                            "width": "100%",
                          },
                        ],
                        Object {
                          "flex": 0,
                          "height": "100%",
                          "width": "100%",
                        },
                      ]
                    }
                    vbHeight={19.054}
                    vbWidth={18.225}
                    width="100%"
                  >
                    <RNSVGGroup>
                      <RNSVGGroup
                        matrix={
                          Array [
                            -1,
                            1.2246467991473532e-16,
                            -1.2246467991473532e-16,
                            -1,
                            17.225,
                            17.639,
                          ]
                        }
                        name="Seta_-_icon"
                      >
                        <RNSVGPath
                          d="M7.5,18H23.725"
                          fill={null}
                          matrix={
                            Array [
                              1,
                              0,
                              0,
                              1,
                              -7.5,
                              -9.887,
                            ]
                          }
                          name="Caminho_68"
                          propList={
                            Array [
                              "fill",
                              "stroke",
                              "strokeWidth",
                              "strokeLinecap",
                              "strokeLinejoin",
                            ]
                          }
                          stroke={4281545523}
                          strokeLinecap={1}
                          strokeLinejoin={1}
                          strokeWidth="2"
                        />
                        <RNSVGPath
                          d="M18,7.5l8.113,8.113L18,23.725"
                          fill={null}
                          matrix={
                            Array [
                              1,
                              0,
                              0,
                              1,
                              -9.887,
                              -7.5,
                            ]
                          }
                          name="Caminho_69"
                          propList={
                            Array [
                              "fill",
                              "stroke",
                              "strokeWidth",
                              "strokeLinecap",
                              "strokeLinejoin",
                            ]
                          }
                          stroke={4281545523}
                          strokeLinecap={1}
                          strokeLinejoin={1}
                          strokeWidth="2"
                        />
                      </RNSVGGroup>
                    </RNSVGGroup>
                  </RNSVGSvgView>
                </RNSVGGroup>
              </RNSVGSvgView>
              <View
                justifyContent="center"
                style={
                  Array [
                    Object {
                      "justifyContent": "center",
                    },
                  ]
                }
              >
                <Text
                  letterSpacing={1.6}
                  style={
                    Array [
                      Object {
                        "letterSpacing": 1.6,
                      },
                    ]
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        alignItems="center"
        alignSelf="center"
        justifyContent="flex-start"
        style={
          Array [
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "justifyContent": "flex-start",
              "width": "50%",
            },
          ]
        }
        width="50%"
      >
        <RNSVGSvgView
          bbHeight={24}
          bbWidth={24}
          color="vermelhoAlerta"
          focusable={false}
          size={24}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Array [
                Object {
                  "color": "#EF1E1E",
                  "height": 24,
                  "width": 24,
                },
              ],
              Object {
                "flex": 0,
                "height": 24,
                "width": 24,
              },
            ]
          }
        >
          <RNSVGGroup>
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight="100%"
              bbWidth="100%"
              focusable={false}
              id="Logo"
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Array [
                    Object {},
                  ],
                  Object {
                    "flex": 0,
                    "height": "100%",
                    "width": "100%",
                  },
                ]
              }
              vbHeight={124.992}
              vbWidth={102.034}
            >
              <RNSVGGroup>
                <RNSVGPath
                  d="M100.065,114.205a115.2,115.2,0,0,1-14.714-10.974,1.1,1.1,0,0,1,0-.964,1.914,1.914,0,0,0-.414-1.849c-2.806-2.99-5.736-5.649-8.644-8.278-2.338-3.043-4.2-6.308-6.4-9.411-1.977-3.352-5.162-8.109-6.39-9.852-1.2-1.736-6.055-7.791-7.8-10.176-2.243-2.862-5.394-6.677-7.815-8.066-2.327-1.344-9.272-5.645-11.783-9.411-1.144-1.842-2.867-8.38-2.623-11.29a12.6,12.6,0,0,1,.9-4.429,46.088,46.088,0,0,1,3.136-5.81c.92-1.39,2.672-2.588,2.471-4.463-.178-1.126.7-2.617-.517-3.378-.368-.094-.691.55-.768.55-.091-.031-.091-.094-.091-.094a16.07,16.07,0,0,1,1.053-5.676c.323-1.307-.285-2.387-.261-3.657.319-.776,2.075-1.917.938-2.331-1.205-.422.053-1.921-1.137-2.052C37.994,2.507,38.251.387,37.092.877c-2.25,1.548-5.629,6.052-6.024,5.9-.384-.17.255-4.225.6-6.263C31.508-.37,30.228.142,29.4.2a7.665,7.665,0,0,1-2.2.063A49.761,49.761,0,0,0,23.788,4.3a41.5,41.5,0,0,0-1.212,4.463c-.138.561-.5,1.691-.544.561-.072-1.081,1.25-4.9,1.525-7.543-.195-.339-.669-.222-.875-.154a10.451,10.451,0,0,0-2.053,1.4,17.051,17.051,0,0,0-1.258,2.358c-.08.38-.08,2.052-.619,1.725-.609-.689,0-1.988,0-2.312.053-.068-.125-.508-.9-.494s-.81-.388-1.164-.18a14.339,14.339,0,0,1-1.486,3.216c-.718,1.4-1.827,3.077-1.881,4.323-.1.441-.46.139-.582.03,0,0-1.562-.663-1.669.411a3.442,3.442,0,0,0,.578,1.751c.38.591,1.817,1.639,1.555,3.005a5.742,5.742,0,0,0-.783.572c-.673,1.134,2.413,6.7,3.287,8.334.035.162-.512.275-.512.275C13.62,26.022,7.254,28.907,5.98,29.8A64.114,64.114,0,0,0,.306,34.97c-.476.4-.365,1.352.118.995a33.091,33.091,0,0,1,11.228-5.717C16.942,28.873,19,32.353,19,32.353s2.531,3.529-.083,11.234c-1.718,5.664-3.949,10.1-3.508,17.3.555,8.5,4.873,13.723,8.362,17.316s13.5,14.386,15.011,15.84c1.5,1.438,2.581,4.323,3.474,5.321a24.772,24.772,0,0,0,3.159,3.333,6.8,6.8,0,0,1,.144,2.154c0,.852-1.4,8.873-1.52,8.982-.3.569-1.44.569-1.44.569a51.155,51.155,0,0,0-6.481-1.939c-.8.113-1.38,1.45-2.117,2.214-.932.889.567,1.8-.609,2.674-.3.234-.209.817-.083,1.009,0,0,.2-.023.6-.456a1.084,1.084,0,0,1,1.035-.309c4.549,1.431,9.989.478,10.456.43.467-.087,2.637,1.242,2.637,1.242,2.049,1.1,4.144,2.052,6.162,3.227.627.241,1.117,1.236,1.832.6.388-.8-.467-1.446-.467-1.446-.153-.328-3.41-2.4-5.139-3.622a14.5,14.5,0,0,0-3.171-3.325c-.193-.768,1.087-4.109,1.407-5.461s.308-4.033,1.03-5.849a11.992,11.992,0,0,1,3.372-2.285,76.542,76.542,0,0,0,7.272,2.636c.863.264,1.619.485,2.383.693-.038.467-.092.952-.092,1.363,0,0-.452,3.13-.582,3.529a2.568,2.568,0,0,1-.491.772c-.311.607-1.455.584-1.455.584a58.567,58.567,0,0,0-6.489-2.06c-.832.43-1.428,1.476-2.132,2.21-.923.885.532,1.823-.639,2.712-.285.181-.228.787-.061.968,0,0,.159,0,.6-.459a1.062,1.062,0,0,1,1.049-.283c4.543,1.465,8.382.283,10.393.52l2.642,1.311c2.06,1.088,4.143,2.1,6.146,3.287.6.23,1.361,1.277,1.836.6.467-.7-.077-1.054-.452-1.416-.407-.41-3.4-2.474-5.105-3.683a17.288,17.288,0,0,0-3.273-3.536c-.087-.411.145-.708.243-1.077.084-.32,1.141-4.244,1.141-4.244s0-.034.038-.064c3.018.986,5.036,1.77,6.005,2.538,2.638,1.928,5.348,3.334,7.435,5.95,3.713,3.393,6.959,6.67,11.441,9.137C94.8,125.8,98.229,125,99.3,124.4c1.061-.561,2.121-1.013,2.03-2.271,2.037-2.734-.969-7.845-1.266-7.924"
                  fill={4293729316}
                  matrix={
                    Array [
                      1,
                      0,
                      0,
                      1,
                      0,
                      0,
                    ]
                  }
                  name="#EF1E1E"
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
      <View
        alignItems="flex-end"
        alignSelf="center"
        flexDirection="row"
        justifyContent="flex-end"
        style={
          Array [
            Object {
              "alignItems": "flex-end",
              "alignSelf": "center",
              "flexDirection": "row",
              "justifyContent": "flex-end",
              "width": "25%",
            },
          ]
        }
        width="25%"
      />
    </View>
  </View>
  <View
    alignItems="center"
    flex={1}
    justifyContent="center"
    mt="xxl"
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "marginTop": 80,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "aspectRatio": 1,
          },
          undefined,
          Object {
            "width": 60,
          },
        ]
      }
    >
      <View
        autoPlay={true}
        cacheComposition={true}
        enableMergePathsAndroidForKitKatAndAbove={false}
        forwardedRef={[Function]}
        loop={true}
        onAnimationFinish={[Function]}
        onLayout={[Function]}
        progress={0}
        resizeMode="contain"
        sourceJson="{\\"v\\":\\"5.5.7\\",\\"meta\\":{\\"g\\":\\"LottieFiles AE 0.1.21\\",\\"a\\":\\"Abigail Ambrogi\\",\\"k\\":\\"Loading, circle, waiting\\",\\"d\\":\\"Simple soft loading animation\\",\\"tc\\":\\"none\\"},\\"fr\\":29.9700012207031,\\"ip\\":0,\\"op\\":50.0000020365418,\\"w\\":800,\\"h\\":800,\\"nm\\":\\"Rounded Loading Animation\\",\\"ddd\\":0,\\"assets\\":[],\\"layers\\":[{\\"ddd\\":0,\\"ind\\":1,\\"ty\\":4,\\"nm\\":\\"Spinning Circle\\",\\"sr\\":1,\\"ks\\":{\\"o\\":{\\"a\\":0,\\"k\\":100,\\"ix\\":11},\\"r\\":{\\"a\\":1,\\"k\\":[{\\"i\\":{\\"x\\":[0.667],\\"y\\":[1]},\\"o\\":{\\"x\\":[0.333],\\"y\\":[0]},\\"t\\":0,\\"s\\":[0]},{\\"t\\":50.0000020365418,\\"s\\":[360]}],\\"ix\\":10},\\"p\\":{\\"a\\":0,\\"k\\":[400,400,0],\\"ix\\":2},\\"a\\":{\\"a\\":0,\\"k\\":[-32.74,-34.232,0],\\"ix\\":1},\\"s\\":{\\"a\\":0,\\"k\\":[100.424,100.424,100],\\"ix\\":6}},\\"ao\\":0,\\"shapes\\":[{\\"ty\\":\\"gr\\",\\"it\\":[{\\"d\\":1,\\"ty\\":\\"el\\",\\"s\\":{\\"a\\":0,\\"k\\":[704.67,704.67],\\"ix\\":2},\\"p\\":{\\"a\\":0,\\"k\\":[0,0],\\"ix\\":3},\\"nm\\":\\"Ellipse Path 1\\",\\"mn\\":\\"ADBE Vector Shape - Ellipse\\",\\"hd\\":false},{\\"ty\\":\\"tm\\",\\"s\\":{\\"a\\":1,\\"k\\":[{\\"i\\":{\\"x\\":[0.667],\\"y\\":[1]},\\"o\\":{\\"x\\":[0.333],\\"y\\":[0]},\\"t\\":10,\\"s\\":[0]},{\\"t\\":50.0000020365418,\\"s\\":[100]}],\\"ix\\":1},\\"e\\":{\\"a\\":1,\\"k\\":[{\\"i\\":{\\"x\\":[0.667],\\"y\\":[1]},\\"o\\":{\\"x\\":[0.333],\\"y\\":[0]},\\"t\\":0,\\"s\\":[0.2]},{\\"t\\":30.0000012219251,\\"s\\":[100]}],\\"ix\\":2},\\"o\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":3},\\"m\\":1,\\"ix\\":2,\\"nm\\":\\"Trim Paths 1\\",\\"mn\\":\\"ADBE Vector Filter - Trim\\",\\"hd\\":false},{\\"ty\\":\\"st\\",\\"c\\":{\\"a\\":0,\\"k\\":[0.6078431372549019,0.6078431372549019,0.6078431372549019,1],\\"ix\\":3},\\"o\\":{\\"a\\":0,\\"k\\":100,\\"ix\\":4},\\"w\\":{\\"a\\":0,\\"k\\":100,\\"ix\\":5},\\"lc\\":2,\\"lj\\":1,\\"ml\\":4,\\"bm\\":0,\\"nm\\":\\"Stroke 1\\",\\"mn\\":\\"ADBE Vector Graphic - Stroke\\",\\"hd\\":false},{\\"ty\\":\\"tr\\",\\"p\\":{\\"a\\":0,\\"k\\":[-32.74,-34.232],\\"ix\\":2},\\"a\\":{\\"a\\":0,\\"k\\":[0,0],\\"ix\\":1},\\"s\\":{\\"a\\":0,\\"k\\":[97.674,97.674],\\"ix\\":3},\\"r\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":6},\\"o\\":{\\"a\\":0,\\"k\\":100,\\"ix\\":7},\\"sk\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":4},\\"sa\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":5},\\"nm\\":\\"Transform\\"}],\\"nm\\":\\"Ellipse 1\\",\\"np\\":4,\\"cix\\":2,\\"bm\\":0,\\"ix\\":1,\\"mn\\":\\"ADBE Vector Group\\",\\"hd\\":false}],\\"ip\\":0,\\"op\\":50.0000020365418,\\"st\\":0,\\"bm\\":0},{\\"ddd\\":0,\\"ind\\":2,\\"ty\\":4,\\"nm\\":\\"Background Circle\\",\\"sr\\":1,\\"ks\\":{\\"o\\":{\\"a\\":0,\\"k\\":100,\\"ix\\":11},\\"r\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":10},\\"p\\":{\\"a\\":0,\\"k\\":[400,400,0],\\"ix\\":2},\\"a\\":{\\"a\\":0,\\"k\\":[-32.74,-34.232,0],\\"ix\\":1},\\"s\\":{\\"a\\":0,\\"k\\":[100.424,100.424,100],\\"ix\\":6}},\\"ao\\":0,\\"shapes\\":[{\\"ty\\":\\"gr\\",\\"it\\":[{\\"d\\":1,\\"ty\\":\\"el\\",\\"s\\":{\\"a\\":0,\\"k\\":[704.67,704.67],\\"ix\\":2},\\"p\\":{\\"a\\":0,\\"k\\":[0,0],\\"ix\\":3},\\"nm\\":\\"Ellipse Path 1\\",\\"mn\\":\\"ADBE Vector Shape - Ellipse\\",\\"hd\\":false},{\\"ty\\":\\"st\\",\\"c\\":{\\"a\\":0,\\"k\\":[0.6078431372549019,0.6078431372549019,0.6078431372549019,1],\\"ix\\":3},\\"o\\":{\\"a\\":0,\\"k\\":100,\\"ix\\":4},\\"w\\":{\\"a\\":0,\\"k\\":100,\\"ix\\":5},\\"lc\\":2,\\"lj\\":1,\\"ml\\":4,\\"bm\\":0,\\"nm\\":\\"Stroke 1\\",\\"mn\\":\\"ADBE Vector Graphic - Stroke\\",\\"hd\\":false},{\\"ty\\":\\"tr\\",\\"p\\":{\\"a\\":0,\\"k\\":[-32.74,-34.232],\\"ix\\":2},\\"a\\":{\\"a\\":0,\\"k\\":[0,0],\\"ix\\":1},\\"s\\":{\\"a\\":0,\\"k\\":[97.674,97.674],\\"ix\\":3},\\"r\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":6},\\"o\\":{\\"a\\":0,\\"k\\":100,\\"ix\\":7},\\"sk\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":4},\\"sa\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":5},\\"nm\\":\\"Transform\\"}],\\"nm\\":\\"Ellipse 1\\",\\"np\\":3,\\"cix\\":2,\\"bm\\":0,\\"ix\\":1,\\"mn\\":\\"ADBE Vector Group\\",\\"hd\\":false}],\\"ip\\":50.0000020365418,\\"op\\":50.0000020365418,\\"st\\":0,\\"bm\\":0,\\"hidden\\":0}],\\"markers\\":[]}"
        speed={1}
        style={
          Object {
            "aspectRatio": 1,
            "height": "100%",
            "width": 60,
          }
        }
      />
    </View>
  </View>
</RCTSafeAreaView>
`;
